### ISSUE TITLE
Ratcheting timeouts

### ISSUE THREAD

**Issue by diziara** [WaltWh/LongSphinx#99]
If using a `!timeout @user` command with the bot, each time it's used on the same user, it would be for a longer period of time.

If long enough between uses, it should reset to the starting value, or at least step down.

Can access to this feature be available to anyone with the timeout permission, instead of just admins?

**Comment by WaltWh** [comment]
This can also be available as a right-click command, on users or messages or both.

**Comment by WaltWh** [comment]
What should the ratcheting values be? Some options:

1. Multiply by 1.5, rounding up (all values in minutes):
5, 8, 12, 18, 27, 41, 62, 93

2. Multiply by 2:
5, 10, 20, 40, 80, 160, 320, 640

3. Triangular numbers (but x5):
5, 15, 30, 50, 75, 105, 140, 180, 225

4: By units
1 minute, 1 hour, 1 day, 1 week, 1 month, 1 year

5: Fibonnaci (but x5)
5, 10, 15, 25, 40, 65, 105, 170, 275, 445, 720

**Comment by WaltWh** [comment]
And what should the reset duration be?

**Comment by diziara** [comment]
Does the api let you set whatever time values you want? Because I thought we'd just er the defaults that I saw in the discord client ![Capture+_2022-01-17-14-20-16.png](https://user-images.githubusercontent.com/3804952/149843052-02bccbe1-1de2-4561-8f4d-34c6f483e37a.png)

As for the reset, it probably should be proportional to the timeout value or a flat value since the last timeout expired.

I'm not sure what the proportion should be if the former, but we can spitball it? My worry is too low and we'll never use the ratchet, too high and someone could get hit with a high dollar timeout long after the original incident has been resolved?

**Comment by WaltWh** [comment]
Yeah, the API will take whatever we want. Those values are reasonable but then I have to make a lookup instead of just mathing it, boo.

Flat value for the timeout would be way easier to implement; I've set it to 1 week at the moment just to have a number in there, but I'm open to whatever. I do intend for this all to be in the same discipline log as the warnings, so someone with appropriate permissions could actually clear it out if an incident is well and truly resolved to the point that it shouldn't affect future actions.

**Comment by WaltWh** [comment]
there is a question of how the top-level timeout interacts with the reset - i.e. if the top level is one week and the reset is also one week, they're guaranteed to have reset by the time they come back.

**Comment by diziara** [comment]
One week flat timeout sounds good.

If the api takes whatever we want, just straight fibb it. Except replace the first one with thirty seconds, if that's not a pain it the ass without breaking the fibb math to do the normal sequence from there? If that's a pain, then just 1, 1, 2, 3, 5, 8, 13 etc is fine.

**Comment by diziara** [comment]
Reset should be from when the timeout ends, not when it starts.

**Comment by WaltWh** [comment]
but that's harder 😭
